### PR TITLE
Eslint prettier

### PR DESCRIPTION
**Pull request by Spetastian**
DO NOT MERGE THIS YET.

Eslint using prettier rules. There is a problem, we cant use react class member functions that are arrow functions without using the babel-parser with eslint. If we use the babel-parser then the prettier eslint rules will find shit loads of lint errors.

Do we really want to go back to having lot of `.bind()` in our constructors in the react component classes. I say no, we should find another way.

Please try this branch for yourself by running `npm run lint` and see that we get lot of `error  Parsing error: Unexpected token =` because we are using e.g 
```
class Foo extends Component {
...
 someEventHandler = () => {....}
...
}
```

### CODE REVIEW

**Comment by elimisteve** [comment]
> Do we really want to go back to having lot of .bind() in our constructors in the react component classes. I say no, we should find another way.

Agreed.

File: src/App.test.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { Provider } from 'react-redux';
+import {Provider} from 'react-redux';
```
Comment (elimisteve) [inline src/App.test.js:3]: @Spetastian Can you make it so that these spaces between curlies are preserved?

Comment (elimisteve) [reply src/App.test.js:3]: For imports

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "singleQuote": true,
+  "trailingComma": "none",
+  "bracketSpacing": false,
+  "jsxBracketSameLine": true
+}
```
Comment (elimisteve) [inline .prettierrc:6]: @Spetastian Let's require a newline at the end of every file -- .js*, .*rc, etc; everything but binaries

File: src/App.test.js (hunk 1 above)
Comment (Spetastian) [reply src/App.test.js:3]: Yes, I do agree. This is the reason for why we also need eslint aside for prettier. To. Specify our own rules as well.

Comment (Spetastian) [reply src/App.test.js:3]: Yes, I do agree. This is the reason for why we also need eslint aside for prettier. To. Specify our own rules as well.

File: src/components/Dashboard/Dashboard.js [hunk 3]
```diff
@@ -49,4 +47,4 @@ class Dashboard extends Component {
   }
 }
 
-export default connect(null, { getPursuances })(Dashboard);
+export default connect(null, {getPursuances})(Dashboard);
```
Comment (elimisteve) [inline src/components/Dashboard/Dashboard.js:50]: @Spetastian ...and preserve spaces between these curlies.

File: src/components/Content/TaskManager/TaskForm/Suggestions/AssignerSuggestions.js [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
+        return <li {...props}>{suggestionName}</li>;
       })}
     </ul>
-</div>
-)
+  </div>
+);
 
-export default connect(({ taskForm }) => ({ taskForm }), { addSuggestion })(AssignerSuggestions);
+export default connect(({taskForm}) => ({taskForm}), {addSuggestion})(
```
Comment (elimisteve) [inline src/components/Content/TaskManager/TaskForm/Suggestions/AssignerSuggestions.js:32]: @Spetastian ...and these curlies

File: src/App.js [hunk 5]
```diff
@@ -1,28 +1,27 @@
...
       <Router>
         <div className="App">
-          <NavBar 
-            authenticated={ this.props.authenticated } 
-            contributionPoints={ this.props.contributionPoints }
-            username={ this.props.username } 
+          <NavBar
+            authenticated={this.props.authenticated}
+            contributionPoints={this.props.contributionPoints}
+            username={this.props.username}
```
Comment (elimisteve) [inline src/App.js:21]: @Spetastian But eliminating these spaces makes sense to me, but if re-adding the spaces between import curly braces requires re-adding these spaces, that's fine with me.

**Comment by elimisteve** [comment]
@Spetastian So I think the latest plan is to implement this? https://chat.cypherpunkswritecode.com/community/pl/6qh8rgooibbnu8wbfd843aejya

**Pull request closed without merging**